tests/gui/TestGui.cpp

```diff
@@ -1473,7 +1477,90 @@ void TestGui::testDatabaseSettings()
     QTest::keyClick(transformRoundsSpinBox, Qt::Key_Enter);
     QTRY_COMPARE(m_db->kdf()->rounds(), 123456);
 
+    // Test loading default values and setting autosaveDelay
+    triggerAction("actionDatabaseSettings");
+    QVERIFY(autosaveDelayCheckBox->isChecked() == false);
+    autosaveDelayCheckBox->toggle();
+    autosaveDelaySpinBox->setValue(autosaveDelayTestValue);
+    QTest::mouseClick(dbSettingsButtonBox->button(QDialogButtonBox::Ok), Qt::LeftButton);
+    QTRY_COMPARE(m_db->metadata()->autosaveDelayMin(), autosaveDelayTestValue);
+
     checkSaveDatabase();
+
+    // Test loading autosaveDelay non-default values
+    triggerAction("actionDatabaseSettings");
+    QTRY_COMPARE(autosaveDelayCheckBox->isChecked(), true);
+    QTRY_COMPARE(autosaveDelaySpinBox->value(), autosaveDelayTestValue);
+    QTest::mouseClick(dbSettingsButtonBox->button(QDialogButtonBox::Cancel), Qt::LeftButton);
+
+    // test autosave delay
+
+    // 1 init
+    config()->set(Config::AutoSaveAfterEveryChange, true);
+    QSignalSpy writeDbSignalSpy(m_db.data(), &Database::databaseSaved);
+
+    // 2 create new entries
+
+    // 2.a) Click the new entry button and set the title
+    auto* entryNewAction = m_mainWindow->findChild<QAction*>("actionEntryNew");
+    QVERIFY(entryNewAction->isEnabled());
+
+    auto* toolBar = m_mainWindow->findChild<QToolBar*>("toolBar");
+    QVERIFY(toolBar);
+
+    QWidget* entryNewWidget = toolBar->widgetForAction(entryNewAction);
+    QVERIFY(entryNewWidget->isVisible());
+    QVERIFY(entryNewWidget->isEnabled());
```

Comment:
Up to you, but I think to verify visibility/enabled/etc of not related to your tests components is out of scope. For example, the `entryNewAction` visibility was tested before.

Reply:
seems redundant, removing on my next push